### ISSUE TITLE
[cargo fix] `identity_conversion` removes parenthesis with inner expression, that needs them.

### ISSUE THREAD

**Issue by soruh** [rust-lang/rust-clippy#4750]
When running `cargo fix --clippy -Z unstable-options` the following happens:

This
```rust
let int_c = i32::from(int_a + int_b) * 3;
```
will be turned into this
```rust
let int_c = int_a + int_b * 3;
```
, which is obviously incorrect.
It should instead be turned into this:
```rust
let int_c = (int_a + int_b) * 3;
```

Only tested on this version:
```bash
$ cargo clippy -V
clippy 0.0.212 (e8d5a9e 2019-10-22)
$ cargo -V
cargo 1.40.0-nightly (3a9abe3f0 2019-10-15)
$ rustc -V
rustc 1.40.0-nightly (4a8c5b20c 2019-10-23)
```

**Comment by Areredify** [comment]
Is it ok to just always emit `(...)` regardless of whether they are needed?

**Comment by flip1995** [comment]
There is the `unused_parens` lint, that would deal with this afterwards. But there already is 

https://github.com/rust-lang/rust-clippy/blob/b4f1769734b6204fc6bece8556b7b80a7683271e/clippy_lints/src/utils/sugg.rs#L22-L30

which can be helpful on fixing this issue.
